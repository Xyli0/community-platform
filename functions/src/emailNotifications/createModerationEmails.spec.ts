import { FirebaseEmulatedTest } from '../test/Firebase/emulator'
import { DB_ENDPOINTS, IUserDB } from '../models'
import {
  HOW_TO_APPROVAL_SUBJECT,
  HOW_TO_NEEDS_IMPROVEMENTS_SUBJECT,
  HOW_TO_REJECTED_SUBJECT,
  HOW_TO_SUBMISSION_SUBJECT,
  MAP_PIN_APPROVAL_SUBJECT,
  MAP_PIN_NEEDS_IMPROVEMENTS_SUBJECT,
  MAP_PIN_REJECTED_SUBJECT,
  MAP_PIN_SUBMISSION_SUBJECT,
} from './templates'
import { setMockHowto } from '../emulator/seed/content-generate'
import {
  createHowtoModerationEmail,
  createMapPinModerationEmail,
  handleModerationUpdate,
} from './createModerationEmails'
import { PP_SIGNOFF } from './constants'

jest.mock('../Firebase/auth', () => ({
  firebaseAuth: {
    getUser: () => ({
      email: 'test@test.com',
    }),
  },
}))

jest.mock('../config/config', () => ({
  CONFIG: {
    deployment: {
      site_url: 'https://community.preciousplastic.com',
    },
  },
}))

const userFactory = (_id: string, user: Partial<IUserDB> = {}): IUserDB =>
  ({
    _id,
    _authID: _id,
    ...user,
  } as IUserDB)

describe('Create howto moderation emails', () => {
  const db = FirebaseEmulatedTest.admin.firestore()

  beforeEach(async () => {
    await FirebaseEmulatedTest.clearFirestoreDB()
    await FirebaseEmulatedTest.seedFirestoreDB('emails')

    await FirebaseEmulatedTest.seedFirestoreDB('users', [
      userFactory('user_1', {
        displayName: 'User 1',
        userName: 'user_1',
        userRoles: ['beta-tester'],
      }),
      userFactory('user_2', {
        displayName: 'User 2',
        userName: 'user_2',
      }),
    ])

    await FirebaseEmulatedTest.seedFirestoreDB('howtos')
  })

  afterAll(async () => {
    await FirebaseEmulatedTest.clearFirestoreDB()
  })

  it('Creates an email for an accepted howto', async () => {
    const howtoApproved = await setMockHowto({ uid: 'user_1' })
    const howtoAwaitingModeration = {
      ...howtoApproved,
      moderation: 'awaiting-moderation',
    }
    const change = FirebaseEmulatedTest.mockFirestoreChangeObject(
      howtoAwaitingModeration,
      howtoApproved,
      'howtos',
      howtoApproved._id,
    )

    await handleModerationUpdate(change, createHowtoModerationEmail)

    // Only one approved howto email should have been created
    const countSnapshot = await db.collection(DB_ENDPOINTS.emails).count().get()
    expect(countSnapshot.data().count).toEqual(1)

    const querySnapshot = await db.collection(DB_ENDPOINTS.emails).get()
    querySnapshot.forEach((doc) => {
      const {
        message: { html, subject },
        to,
      } = doc.data()
      expect(html).toMatchSnapshot()
      expect(subject).toBe(HOW_TO_APPROVAL_SUBJECT)
      // Check that the email contains the correct user name
      expect(html).toContain('Hey User 1')
      // Check that the email contains the correct howto title
      expect(html).toContain(
        'Huzzah! Your How-To Mock Howto has been approved.',
      )
      // Check that the email contains the correct howto link
      expect(html).toContain(
        'https://community.preciousplastic.com/how-to/00_user_1_howto',
      )
      // Check that the email contains the correct PP signoff
      expect(html).toContain(PP_SIGNOFF)
      expect(to).toBe('test@test.com')
    })
  })

  it('Creates an email for a howto awaiting moderation', async () => {
    const howtoRejected = await setMockHowto({ uid: 'user_1' }, 'rejected')
    const howtoAwaitingModeration = {
      ...howtoRejected,
      moderation: 'awaiting-moderation',
    }
    const change = FirebaseEmulatedTest.mockFirestoreChangeObject(
      howtoRejected,
      howtoAwaitingModeration,
      'howtos',
      howtoAwaitingModeration._id,
    )

    await handleModerationUpdate(change, createHowtoModerationEmail)

    // Only one submitted howto email should have been created
    const countSnapshot = await db.collection(DB_ENDPOINTS.emails).count().get()
    expect(countSnapshot.data().count).toEqual(1)

    const querySnapshot = await db.collection(DB_ENDPOINTS.emails).get()
    querySnapshot.forEach((doc) => {
      const {
        message: { html, subject },
        to,
      } = doc.data()
      expect(html).toMatchSnapshot()
      expect(subject).toBe(HOW_TO_SUBMISSION_SUBJECT)
      // Check that the email contains the correct user name
      expect(html).toContain('Hey User 1')
      // Check that the email contains the correct howto title
      expect(html).toContain(
        `Huzzah! Your How-To Mock Howto has been submitted.`,
      )
      // Check that the email contains the correct PP signoff
      expect(html).toContain(PP_SIGNOFF)
      expect(to).toBe('test@test.com')
    })
  })

  it('Creates an email for a rejected howto', async () => {
    const howtoAwaitingModeration = await setMockHowto(
      { uid: 'user_1' },
      'awaiting-moderation',
    )
    const howtoRejected = {
      ...howtoAwaitingModeration,
      moderation: 'rejected',
    }
    const change = FirebaseEmulatedTest.mockFirestoreChangeObject(
      howtoAwaitingModeration,
      howtoRejected,
      'howtos',
      howtoAwaitingModeration._id,
    )

    await handleModerationUpdate(change, createHowtoModerationEmail)

    // Only one rejected howto email should have been created
    const countSnapshot = await db.collection(DB_ENDPOINTS.emails).count().get()
    expect(countSnapshot.data().count).toEqual(1)

    const querySnapshot = await db.collection(DB_ENDPOINTS.emails).get()
    querySnapshot.forEach((doc) => {
      const {
        message: { html, subject },
        to,
      } = doc.data()
      expect(subject).toBe(HOW_TO_REJECTED_SUBJECT)
      // Check that the email contains the correct user name
      expect(html).toContain('Hey User 1')
      // Check that the email contains the correct howto title
      expect(html).toContain(`Thank you for submitting your How-To, Mock Howto`)
      expect(html).toContain(
        'However, after reviewing your submission, we feel that',
      )
      // Check that the email contains the correct PP signoff
      expect(html).toContain(PP_SIGNOFF)
      expect(to).toBe('test@test.com')
    })
  })

  it('Creates an email for a howto that needs improvements', async () => {
    const howtoAwaitingModeration = await setMockHowto(
      { uid: 'user_1' },
      'awaiting-moderation',
    )
    const MOCK_HOW_TO_MODERATION_COMMENT = 'Mock how to moderation comment'
    const howtoNeedsImprovements = {
      ...howtoAwaitingModeration,
      moderation: 'improvements-needed',
      comments: MOCK_HOW_TO_MODERATION_COMMENT,
    }
    const change = FirebaseEmulatedTest.mockFirestoreChangeObject(
      howtoAwaitingModeration,
      howtoNeedsImprovements,
      'howtos',
      howtoAwaitingModeration._id,
    )

    await handleModerationUpdate(change, createHowtoModerationEmail)

    // Only one needs improvements howto email should have been created
    const countSnapshot = await db.collection(DB_ENDPOINTS.emails).count().get()
    expect(countSnapshot.data().count).toEqual(1)

    const querySnapshot = await db.collection(DB_ENDPOINTS.emails).get()
    querySnapshot.forEach((doc) => {
      const {
        message: { html, subject },
        to,
      } = doc.data()
      expect(subject).toBe(HOW_TO_NEEDS_IMPROVEMENTS_SUBJECT)
      // Check that the email contains the correct user name
      expect(html).toContain('Hey User 1')
      // Check that the email contains the correct howto title
      expect(html).toContain(
        `Your How-To, Mock Howto, needs more work before we can publish it.`,
      )
      expect(html).toContain(MOCK_HOW_TO_MODERATION_COMMENT)
      // Check that the email contains the correct PP signoff
      expect(html).toContain(PP_SIGNOFF)
      expect(to).toBe('test@test.com')
    })
  })

  // Remove this test once released to all users.
  it('Does not create an email for people who are not beta testers', async () => {
    const howtoApproved = await setMockHowto({ uid: 'user_2' })
    const howtoAwaitingModeration = {
      ...howtoApproved,
      moderation: 'awaiting-moderation',
    }
    const change = FirebaseEmulatedTest.mockFirestoreChangeObject(
      howtoAwaitingModeration,
      howtoApproved,
      'howtos',
      howtoApproved._id,
    )

    await handleModerationUpdate(change, createHowtoModerationEmail)

    // No new emails should have been created
    const countSnapshot = await db.collection(DB_ENDPOINTS.emails).count().get()
    expect(countSnapshot.data().count).toEqual(0)
  })

  it('Does not create an email for non-approved howtos', async () => {
    const howtoApproved = await setMockHowto({ uid: 'user_1' })
    const howtoDraft = {
      ...howtoApproved,
      moderation: 'draft',
    }
    const change = FirebaseEmulatedTest.mockFirestoreChangeObject(
      howtoApproved,
      howtoDraft,
      'howtos',
      howtoApproved._id,
    )

    await handleModerationUpdate(change, createHowtoModerationEmail)

    // No new emails should have been created
    const countSnapshot = await db.collection(DB_ENDPOINTS.emails).count().get()
    expect(countSnapshot.data().count).toEqual(0)
  })
})

describe('Create map pin moderation emails', () => {
  const db = FirebaseEmulatedTest.admin.firestore()

  beforeEach(async () => {
    await FirebaseEmulatedTest.clearFirestoreDB()
    await FirebaseEmulatedTest.seedFirestoreDB('emails')

    await FirebaseEmulatedTest.seedFirestoreDB('users', [
      userFactory('user_1', {
        displayName: 'User 1',
        userName: 'user_1',
        userRoles: ['beta-tester'],
      }),
      userFactory('user_2', {
        displayName: 'User 2',
        userName: 'user_2',
      }),
    ])

    await FirebaseEmulatedTest.seedFirestoreDB('mappins')
  })

  afterAll(async () => {
    await FirebaseEmulatedTest.clearFirestoreDB()
  })

  it('Creates an email for an accepted map pin', async () => {
    const mapPinApproved = {
      _id: 'user_1',
      moderation: 'accepted',
    }
    const mapPinAwaitingModeration = {
      _id: 'user_1',
      moderation: 'awaiting-moderation',
    }
    const change = FirebaseEmulatedTest.mockFirestoreChangeObject(
      mapPinAwaitingModeration,
      mapPinApproved,
      'mappins',
      mapPinApproved._id,
    )

    await handleModerationUpdate(change, createMapPinModerationEmail)

    // Only one approved map pin email should have been created
    const countSnapshot = await db.collection(DB_ENDPOINTS.emails).count().get()
    expect(countSnapshot.data().count).toEqual(1)

    const querySnapshot = await db.collection(DB_ENDPOINTS.emails).get()
    querySnapshot.forEach((doc) => {
      const {
        message: { html, subject },
        to,
      } = doc.data()
      expect(html).toMatchSnapshot()
      expect(subject).toBe(MAP_PIN_APPROVAL_SUBJECT)
      // Check that the email contains the correct user name
      expect(html).toContain('Hey User 1')
      // Check that the email contains the correct map pin link
      expect(html).toContain(
        `https://community.preciousplastic.com/map#${mapPinApproved._id}`,
      )
      // Check that the email contains the correct PP signoff
      expect(html).toContain(PP_SIGNOFF)
      expect(to).toBe('test@test.com')
    })
  })

  it('Creates an email for a map pin awaiting moderation', async () => {
    const mapPinRejected = {
      _id: 'user_1',
      moderation: 'rejected',
    }
    const mapPinAwaitingModeration = {
      _id: 'user_1',
      moderation: 'awaiting-moderation',
    }
    const change = FirebaseEmulatedTest.mockFirestoreChangeObject(
      mapPinRejected,
      mapPinAwaitingModeration,
      'mappins',
      mapPinAwaitingModeration._id,
    )

    await handleModerationUpdate(change, createMapPinModerationEmail)

    // Only one submitted map pin email should have been created
    const countSnapshot = await db.collection(DB_ENDPOINTS.emails).count().get()
    expect(countSnapshot.data().count).toEqual(1)

    const querySnapshot = await db.collection(DB_ENDPOINTS.emails).get()
    querySnapshot.forEach((doc) => {
      const {
        message: { html, subject },
        to,
      } = doc.data()
      expect(html).toMatchSnapshot()
      expect(subject).toBe(MAP_PIN_SUBMISSION_SUBJECT)
      // Check that the email contains the correct user name
      expect(html).toContain('Hey User 1')
      // Check that the email contains the correct title
      expect(html).toContain('Your map pin has been submitted.')
      // Check that the email contains the correct PP signoff
      expect(html).toContain(PP_SIGNOFF)
      expect(to).toBe('test@test.com')
    })
  })

  it('Creates an email for a rejected map pin', async () => {
    const mapPinAwaitingModeration = {
      _id: 'user_1',
      moderation: 'awaiting-moderation',
    }
    const mapPinRejected = {
      _id: 'user_1',
      moderation: 'rejected',
    }
    const change = FirebaseEmulatedTest.mockFirestoreChangeObject(
      mapPinAwaitingModeration,
      mapPinRejected,
      'mappins',
      mapPinRejected._id,
    )

    await handleModerationUpdate(change, createMapPinModerationEmail)

    // Only one rejected map pin email should have been created
    const countSnapshot = await db.collection(DB_ENDPOINTS.emails).count().get()
    expect(countSnapshot.data().count).toEqual(1)

    const querySnapshot = await db.collection(DB_ENDPOINTS.emails).get()
    querySnapshot.forEach((doc) => {
      const {
        message: { html, subject },
        to,
      } = doc.data()
      expect(subject).toBe(MAP_PIN_REJECTED_SUBJECT)
      // Check that the email contains the correct user name
      expect(html).toContain('Hey User 1')
      expect(html).toContain(
        'Thank you for applying to be on the Precious Plastic Map.',
      )
      // Check that the email contains the correct guidelines link
      expect(html).toContain(
        `https://drive.google.com/file/d/1fXTtBbzgCO0EL6G9__aixwqc-Euqgqnd/view`,
      )
      // Check that the email contains the correct PP signoff
      expect(html).toContain(PP_SIGNOFF)
      expect(to).toBe('test@test.com')
    })
  })

  it('Creates an email for a needs improvements map pin', async () => {
    const mapPinAwaitingModeration = {
      _id: 'user_1',
      moderation: 'awaiting-moderation',
    }
    const MOCK_MAP_PIN_MODERATION_COMMENT = 'Mock map pin moderation comment'
    const mapPinNeedsImprovements = {
      _id: 'user_1',
      moderation: 'improvements-needed',
      comments: MOCK_MAP_PIN_MODERATION_COMMENT,
    }
    const change = FirebaseEmulatedTest.mockFirestoreChangeObject(
      mapPinAwaitingModeration,
      mapPinNeedsImprovements,
      'mappins',
      mapPinNeedsImprovements._id,
    )

    await handleModerationUpdate(change, createMapPinModerationEmail)

    // Only one needs improvements map pin email should have been created
    const countSnapshot = await db.collection(DB_ENDPOINTS.emails).count().get()
    expect(countSnapshot.data().count).toEqual(1)

    const querySnapshot = await db.collection(DB_ENDPOINTS.emails).get()
    querySnapshot.forEach((doc) => {
      const {
        message: { html, subject },
        to,
      } = doc.data()
      expect(subject).toBe(MAP_PIN_NEEDS_IMPROVEMENTS_SUBJECT)
      // Check that the email contains the correct user name
      expect(html).toContain('Hey User 1')
      expect(html).toContain(
        "We're really happy to see you on the Precious Plastic Map",
      )
      // Check that the email contains the correct guidelines link
      expect(html).toContain(
        `https://drive.google.com/file/d/1fXTtBbzgCO0EL6G9__aixwqc-Euqgqnd/view`,
      )
      expect(html).toContain(MOCK_MAP_PIN_MODERATION_COMMENT)
      // Check that the email contains the correct PP signoff
      expect(html).toContain(PP_SIGNOFF)
      expect(to).toBe('test@test.com')
    })
  })

  // Remove this test once released to all users.
  it('Does not create an email for people who are not beta testers', async () => {
    const mapPinApproved = {
      _id: 'user_2',
      moderation: 'accepted',
    }
    const mapPinAwaitingModeration = {
      _id: 'user_2',
      moderation: 'awaiting-moderation',
    }
    const change = FirebaseEmulatedTest.mockFirestoreChangeObject(
      mapPinAwaitingModeration,
      mapPinApproved,
      'mappins',
      mapPinApproved._id,
    )

    await handleModerationUpdate(change, createMapPinModerationEmail)

    // No new emails should have been created
    const countSnapshot = await db.collection(DB_ENDPOINTS.emails).count().get()
    expect(countSnapshot.data().count).toEqual(0)
  })

  it('Does not create an email for non-approved map pins', async () => {
    const mapPinDraft = {
      _id: 'user_1',
      moderation: 'draft',
    }
    const mapPinAwaitingModeration = {
      _id: 'user_1',
      moderation: 'awaiting-moderation',
    }
    const change = FirebaseEmulatedTest.mockFirestoreChangeObject(
      mapPinAwaitingModeration,
      mapPinDraft,
      'mappins',
      mapPinDraft._id,
    )

    await handleModerationUpdate(change, createMapPinModerationEmail)

    // No new emails should have been created
    const countSnapshot = await db.collection(DB_ENDPOINTS.emails).count().get()
    expect(countSnapshot.data().count).toEqual(0)
  })
})
