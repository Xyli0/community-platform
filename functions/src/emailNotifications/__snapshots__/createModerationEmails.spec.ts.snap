// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create howto moderation emails Creates an email for a howto awaiting moderation 1`] = `
"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html>
  <head>
    <meta http-equiv="Content-Type" content="text/html charset=UTF-8" />
    <style>
      #body {
        font-family: Arial, Helvetica, sans-serif;
        font-size: 14px;
        color: #000000;
      }
      #body a {
        color: inherit;
      }
      .border {
        border: 3px solid black;
        border-radius: 10px;
        padding: 4% 0;
        margin: 4% 0;
        width: 550px;
      }
      table {
        width: 100%;
        border: 0;
      }
      .settings-table-container,
      .project-image-table-container {
        margin-bottom: 8%;
      }

      .greeting-container,
      .main-container {
        margin-bottom: 8%;
      }
      .greeting-container,
      .main-container,
      .closing-containter {
        margin-left: 8%;
        margin-right: 8%;
      }
      @media only screen and (max-width: 550px) {
        .table-container {
          width: 100%;
        }
      }
    </style>
  </head>
  <body id="body">
    <table class="email-table-container">
      <tr>
        <td align="center">
          <div class="border">
            <table class="project-image-table-container">
              <tr>
                <td align="center">
                  <img
                    width="144"
                    alt=""
                    src="http://cdn.mcauto-images-production.sendgrid.net/cb5a685b085b7e7c/f48e2455-6dbb-4102-bde9-2ee07d3008c4/859x860.png"
                  />
                </td>
              </tr>
            </table>

            <div align="left" class="greeting-container">
              <p>Hey User 1</p>
              <p>Huzzah! Your How-To Mock Howto has been submitted.</p>
            </div>
            <div align="left" class="main-container">
              <p>
                Our team is currently reviewing it and will get back to you
                within a few business days.
              </p>
              <p>
                Thank you for taking the time, the community will benefit from
                this knowledge and more plastic will be recycled thanks to YOU.
              </p>
            </div>
            <div align="left" class="closing-containter">
              <p>Cheers,</p>
              <p>The Precious Plastic Team</p>
            </div>
          </div>
          <table class="settings-table-container">
            <tr>
              <td align="center">
                <div class="notifications">
                  Manage your notifications
                  <a href="https://community.preciousplastic.com/settings">
                    here</a
                  >
                </div>
              </td>
            </tr>
          </table>
        </td>
      </tr>
    </table>
  </body>
</html>
"
`;

exports[`Create howto moderation emails Creates an email for an accepted howto 1`] = `
"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html>
  <head>
    <meta http-equiv="Content-Type" content="text/html charset=UTF-8" />
    <style>
      #body {
        font-family: Arial, Helvetica, sans-serif;
        font-size: 14px;
        color: #000000;
      }
      #body a {
        color: inherit;
      }
      .border {
        border: 3px solid black;
        border-radius: 10px;
        padding: 4% 0;
        margin: 4% 0;
        width: 550px;
      }
      table {
        width: 100%;
        border: 0;
      }
      .settings-table-container,
      .project-image-table-container {
        margin-bottom: 8%;
      }

      .greeting-container,
      .main-container {
        margin-bottom: 8%;
      }
      .greeting-container,
      .main-container,
      .closing-containter {
        margin-left: 8%;
      }
      @media only screen and (max-width: 550px) {
        .table-container {
          width: 100%;
        }
      }
    </style>
  </head>
  <body id="body">
    <table class="email-table-container">
      <tr>
        <td align="center">
          <div class="border">
            <table class="project-image-table-container">
              <tr>
                <td align="center">
                  <img
                    width="144"
                    alt=""
                    src="http://cdn.mcauto-images-production.sendgrid.net/cb5a685b085b7e7c/f48e2455-6dbb-4102-bde9-2ee07d3008c4/859x860.png"
                  />
                </td>
              </tr>
            </table>

            <div align="left" class="greeting-container">
              <p>Hey User 1</p>
              <p>Huzzah! Your How-To Mock Howto has been approved.</p>
            </div>
            <div align="left" class="main-container">
              <p>
                It is visible on the community platform
                <a
                  href="https://community.preciousplastic.com/how-to/00_user_1_howto"
                  >here</a
                >.
              </p>
              <p>Users are now be able to comment and mark it as useful.</p>
            </div>
            <div align="left" class="closing-containter">
              <p>Thanks for sharing your knowledge!</p>
              <p>The Precious Plastic Team</p>
            </div>
          </div>
          <table class="settings-table-container">
            <tr>
              <td align="center">
                <div class="notifications">
                  Manage your notifications
                  <a href="https://community.preciousplastic.com/settings">
                    here</a
                  >
                </div>
              </td>
            </tr>
          </table>
        </td>
      </tr>
    </table>
  </body>
</html>
"
`;

exports[`Create map pin moderation emails Creates an email for a map pin awaiting moderation 1`] = `
"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html>
  <head>
    <meta http-equiv="Content-Type" content="text/html charset=UTF-8" />
    <style>
      #body {
        font-family: Arial, Helvetica, sans-serif;
        font-size: 14px;
        color: #000000;
      }
      #body a {
        color: inherit;
      }
      .border {
        border: 3px solid black;
        border-radius: 10px;
        padding: 4% 0;
        margin: 4% 0;
        width: 550px;
      }
      table {
        width: 100%;
        border: 0;
      }
      .settings-table-container,
      .project-image-table-container {
        margin-bottom: 8%;
      }

      .greeting-container,
      .main-container {
        margin-bottom: 8%;
      }
      .greeting-container,
      .main-container,
      .closing-containter {
        margin-left: 8%;
        margin-right: 8%;
      }
      @media only screen and (max-width: 550px) {
        .table-container {
          width: 100%;
        }
      }
    </style>
  </head>
  <body id="body">
    <table class="email-table-container">
      <tr>
        <td align="center">
          <div class="border">
            <table class="project-image-table-container">
              <tr>
                <td align="center">
                  <img
                    width="144"
                    alt=""
                    src="http://cdn.mcauto-images-production.sendgrid.net/cb5a685b085b7e7c/f48e2455-6dbb-4102-bde9-2ee07d3008c4/859x860.png"
                  />
                </td>
              </tr>
            </table>

            <div align="left" class="greeting-container">
              <p>Hey User 1</p>
              <p>Your map pin has been submitted.</p>
            </div>
            <div align="left" class="main-container">
              <p>
                Our team is currently reviewing it and will get back to you
                within a few business days.
              </p>
            </div>
            <div align="left" class="closing-containter">
              <p>Cheers,</p>
              <p>The Precious Plastic Team</p>
            </div>
          </div>
          <table class="settings-table-container">
            <tr>
              <td align="center">
                <div class="notifications">
                  Manage your notifications
                  <a href="https://community.preciousplastic.com/settings">
                    here</a
                  >
                </div>
              </td>
            </tr>
          </table>
        </td>
      </tr>
    </table>
  </body>
</html>
"
`;

exports[`Create map pin moderation emails Creates an email for an accepted map pin 1`] = `
"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html>
  <head>
    <meta http-equiv="Content-Type" content="text/html charset=UTF-8" />
    <style>
      #body {
        font-family: Arial, Helvetica, sans-serif;
        font-size: 14px;
        color: #000000;
      }
      #body a {
        color: inherit;
      }
      .border {
        border: 3px solid black;
        border-radius: 10px;
        padding: 4% 0;
        margin: 4% 0;
        width: 550px;
      }
      table {
        width: 100%;
        border: 0;
      }
      .settings-table-container,
      .project-image-table-container {
        margin-bottom: 8%;
      }

      .greeting-container,
      .main-container {
        margin-bottom: 8%;
      }
      .greeting-container,
      .main-container,
      .closing-containter {
        margin-left: 8%;
        margin-right: 8%;
      }
      @media only screen and (max-width: 550px) {
        .table-container {
          width: 100%;
        }
      }
    </style>
  </head>
  <body id="body">
    <table class="email-table-container">
      <tr>
        <td align="center">
          <div class="border">
            <table class="project-image-table-container">
              <tr>
                <td align="center">
                  <img
                    width="144"
                    alt=""
                    src="http://cdn.mcauto-images-production.sendgrid.net/cb5a685b085b7e7c/f48e2455-6dbb-4102-bde9-2ee07d3008c4/859x860.png"
                  />
                </td>
              </tr>
            </table>

            <div align="left" class="greeting-container">
              <p>Hey User 1</p>
              <p>
                Huzzah! We are happy to inform you that your map pin has been
                approved and is now visible!
              </p>
            </div>
            <div align="left" class="main-container">
              <p>
                It is visible on the community platform map
                <a href="https://community.preciousplastic.com/map#user_1"
                  >here</a
                >.
              </p>
            </div>
            <div align="left" class="closing-containter">
              <p>Thanks,</p>
              <p>The Precious Plastic Team</p>
            </div>
          </div>
          <table class="settings-table-container">
            <tr>
              <td align="center">
                <div class="notifications">
                  Manage your notifications
                  <a href="https://community.preciousplastic.com/settings">
                    here</a
                  >
                </div>
              </td>
            </tr>
          </table>
        </td>
      </tr>
    </table>
  </body>
</html>
"
`;
